### PR TITLE
[4.2] Convert log type field to fancy select

### PR DESCRIPTION
**Pull request by laoneo**
Pull Request for Issue #22627 and alternative for #22699.

### Summary of Changes
Converts the log type field into a fancy select, so all events can be disabled.

### Testing Instructions
- Go to /administrator/index.php?option=com_config&view=component&component=com_actionlogs
- Remove all entries from the "Events To Log" field
- Click save

### Actual result BEFORE applying this Pull Request
All elements are selected.

### Expected result AFTER applying this Pull Request
None is selected.

### CODE REVIEW

**Comment by Kostelano** [comment]
Go to the user profile to the activity log tab, where you can also define a list of extensions for sending notifications.

Screenshot_1

**Comment by brianteeman** [comment]
It wuld be even better if it was like 
image

but if its a choice between the current checkboxes and this list I would stick with the checkboxes. Thye might look ugly but they are much easier to use and you can easily see what is selected without accidentally unleselcting everything.

**Comment by laoneo** [comment]
@Kostelano nice catch. Thanks for the hint. Adapted.

**Comment by Kostelano** [comment]
I have tested this item :white_check_mark: successfully on 57447f8c25ecdfe2a12d5066383e7694ce1d3dd1<br /><br />I will note that everything works. But in part I will support Brian - it seems to me that it is logical to have an ALL button in the drop-down list so that all N extensions are not selected and that it is displayed when selecting ALL extensions with a single click. And, for example, when the ALL button is deleted with a cross, the line remains empty, i.e. no component would be selected.

If something similar can be implemented in this PR, it would be great. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37446">issues.joomla.org/tracker/joomla-cms/37446</a>.</sub>

**Comment by laoneo** [comment]
What I do in DPCalendar is that the default option is "All". But then the user can select individual items, but when the all entry is selected beside individual items then "All" wins. That would be a way to go for me.

**Comment by brianteeman** [comment]
I am not in favour of this change for the reasons I stated above. Long multiple select lists are terrible for usability. It is far too easy to unselect your current selection when scrolling through the list. We already have a solution for this and we should use it instead of inventing another

**Comment by laoneo** [comment]
Thanks for sharing your opinion.

File: administrator/components/com_actionlogs/src/Field/LogtypeField.php [hunk 1]
```diff
@@ -23,7 +23,7 @@
  *
  * @since  3.9.0
  */
-class LogtypeField extends CheckboxesField
+class LogtypeField extends ListField
```
Comment (Quy) [inline administrator/components/com_actionlogs/src/Field/LogtypeField.php:26]: Line 53 can be removed and update line 59.

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 57447f8c25ecdfe2a12d5066383e7694ce1d3dd1<br /><br />fancy-list rocks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37446">issues.joomla.org/tracker/joomla-cms/37446</a>.</sub>

**Comment by laoneo** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37446">issues.joomla.org/tracker/joomla-cms/37446</a>.</sub>

**Comment by brianteeman** [comment]
sorry but as you made changes to the code it cannot be rtc and must be retested

**Comment by Kostelano** [comment]
I have tested this item :white_check_mark: successfully on 35034eacaaf5a84318c3cd9c49eabd186182dfe0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37446">issues.joomla.org/tracker/joomla-cms/37446</a>.</sub>

File: administrator/components/com_actionlogs/src/Field/LogtypeField.php [hunk 2]
```diff
@@ -49,14 +49,13 @@ public function getOptions()
...
 		$extensions = $db->setQuery($query)->loadColumn();
 
-		$options = array();
-		$tmp     = array('checked' => true);
+		$options = [];
 
 		foreach ($extensions as $extension)
 		{
 			ActionlogsHelper::loadTranslationFiles($extension);
 			$option                                                                            = HTMLHelper::_('select.option', $extension, Text::_($extension));
```
Comment (Quy) [inline administrator/components/com_actionlogs/src/Field/LogtypeField.php]: This variable is not necessary and can be combined with the next line.

**Comment by roland-d** [comment]
@laoneo Could you have a look at the question by @Quy and update the branch?

**Comment by laoneo** [comment]
back to pending<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37446">issues.joomla.org/tracker/joomla-cms/37446</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 1c31739223797172103e534baba0b1ebc2044ff4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37446">issues.joomla.org/tracker/joomla-cms/37446</a>.</sub>

**Comment by Kostelano** [comment]
I have tested this item :white_check_mark: successfully on 1c31739223797172103e534baba0b1ebc2044ff4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37446">issues.joomla.org/tracker/joomla-cms/37446</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37446">issues.joomla.org/tracker/joomla-cms/37446</a>.</sub>

**Comment by roland-d** [comment]
Thanks everybody

**Pull request merged**